### PR TITLE
Simplify argument parsing, update docs in ALBERT example

### PR DESCRIPTION
**Pull request by mryab**
This PR updates the documentation in the ALBERT examples to replace all mentions of "the coordinator" with "the training monitor" after #308. Also, the argument parsing logic in `run_trainer.py` is refactored to avoid the current convoluted and error-prone solution of "convert all arguments to a dictionary - pop the arguments you do not need - pass all of the rest to the optimizer". Finally, it contains minor changes in the guide/docstrings meant to improve their clarity.

### CODE REVIEW

File: examples/albert/README.md [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
+and [transformers](https://github.com/huggingface/transformers/) libraries to compute local updates,
+using `hivemind.CollaborativeOptimizer` to exchange information between peers.
+
+## Preparation
 
-### Preparation
 * Install hivemind: `pip install git+https://github.com/learning-at-home/hivemind.git`
 * Dependencies: `pip install -r requirements.txt`
 * Preprocess data: `python tokenize_wikitext103.py`
-* Upload an archive preprocessed data to somewhere volunteers can reach, example: `https://hivemind-data.s3.us-east-2.amazonaws.com/wikitext103_preprocessed.tar`
```
Comment (mryab) [inline examples/albert/README.md:9]: This link is no longer accessible

File: examples/albert/run_training_monitor.py [hunk 2]
```diff
@@ -132,58 +131,60 @@ def upload_checkpoint(self, current_loss):
...
+
             if latest_step != current_step:
                 logger.debug(f"Got metrics from {len(metrics)} peers")
 
                 for i, metrics_for_peer in enumerate(metrics):
                     logger.debug(f"{i} peer {metrics_for_peer}")
+
                 current_step = latest_step
                 alive_peers = 0
-                num_batches = 0
```
Comment (mryab) [inline examples/albert/run_training_monitor.py:182]: This variable is not used

File: examples/albert/README.md (hunk 1 above)
Comment (justheuristic) [reply examples/albert/README.md:9]: Request: can you please upload it to dropbox?

File: examples/albert/README.md [hunk 3]
```diff
@@ -80,31 +104,69 @@ For convenience, you can view (and share!) the learning curves of your collabora
...
+
+For small experiments (3-16 peers, <1GB data), you can use a free-tier file hosting that has a convenient way
+to [download with curl/wget](https://superuser.com/questions/470664/how-to-download-dropbox-files-using-wget-command).
+However, these services are not meant for high load and could ban you for generating too much traffic. If you want to
+scale up, you could either use an S3-like storage from [any](https://aws.amazon.com/s3/)
+[cloud](https://cloud.google.com/storage) [provider](https://cloud.google.com/storage) or host the data
```
Comment (justheuristic) [inline examples/albert/README.md:117]: Google is specified twice. Please change to azure or ycloud

File: examples/albert/README.md [hunk 4]
```diff
@@ -80,31 +104,69 @@ For convenience, you can view (and share!) the learning curves of your collabora
...
+splitting them into chunks and implementing a custom dataloader that can load chunks on the fly. Finally, the most _
+comme il faut_ solution to sharing large datasets is to use [academic torrents](https://academictorrents.com/).
 
 ### run_training_monitor.py
-This peer exists solely to welcome other peers onto the DHT and track learning progress. It requires neither GPU nor high bandwidth, the only prerequisite is that coordinator should have high uptime. If no high uptime server is available, one can also run multiple coordinators on different servers and list all of them as `--initial_peers`. The system will stay up as long as at least one coordinator is available. For short- to mid-term experiments you can host coordinator on a [free-tier VM](https://www.quora.com/Are-there-any-free-online-virtual-machines).
+
+This peer exists solely to welcome other peers onto the DHT and track learning progress. It requires neither GPU nor
+high bandwidth, the only prerequisite is high uptime. If no high uptime server is available, one can also run multiple
+monitors on different servers and list all of them as `--initial_peers`. The system will stay intact as long as at
```
Comment (justheuristic) [inline examples/albert/README.md:126]: request: ~~will stay intact~~ -> will continue training

File: examples/albert/README.md [hunk 5]
```diff
@@ -113,6 +175,11 @@ Here's an example of a full trainer script for Google Colab:
 ```
 
 ### Using IPFS
-If the initial peers for your experiment are located behind NAT and/or you have any trouble with figuring out their public IP addresses and ports, you can set up hivemind to use the [IPFS](https://ipfs.io) network to find the route to your peers automatically. To do this, you should specify the `--use_ipfs` option on all peers (both training monitors and trainers) you are starting.
 
-After that, it is enough to provide only a [libp2p](https://libp2p.io/) peer ID (e.g. `/p2p/XXXX`) for each initial peer. No other information (like IP addresses or TCP/UDP ports) is required.
+If the initial peers for your experiment are located behind NAT and/or you have any trouble with figuring out their
+public IP addresses and ports, you can set up hivemind to use the [IPFS](https://ipfs.io) network to find the route to
+your peers automatically. To do this, you should specify the `--use_ipfs` option on all peers (both training monitors
+and trainers) you are starting.
```
Comment (justheuristic) [inline examples/albert/README.md:182]:
```suggestion
your peers automatically. To do this, you should specify the `--use_ipfs` option on all peers you are starting (both trainers and monitors).
```

**Review by justheuristic — APPROVED** [review]

File: examples/albert/README.md (hunk 4 above)
Comment (mryab) [reply examples/albert/README.md]: Replaced by "will maintain its integrity"

File: examples/albert/README.md (hunk 1 above)
Comment (mryab) [reply examples/albert/README.md:9]: This dataset is very small and gets preprocessed quickly, so hosting it is not really necessary :)

**Comment by yhn112** [comment]
Beautiful 
(But will continue reviewing later this evening)

File: examples/albert/README.md [hunk 6]
```diff
@@ -1,20 +1,31 @@
...
+
+## Preparation
 
-### Preparation
 * Install hivemind: `pip install git+https://github.com/learning-at-home/hivemind.git`
 * Dependencies: `pip install -r requirements.txt`
 * Preprocess data: `python tokenize_wikitext103.py`
-* Upload an archive preprocessed data to somewhere volunteers can reach, example: `https://hivemind-data.s3.us-east-2.amazonaws.com/wikitext103_preprocessed.tar`
-
+* Upload an archive with preprocessed data to someplace volunteers can reach
```
Comment (yhn112) [inline examples/albert/README.md:13]: ..."Or ask your volunteers to run preprocessing themselves, it's rather fast."

File: examples/albert/README.md [hunk 7]
```diff
@@ -1,20 +1,31 @@
...
 
 ## Running an experiment
-- Run the first DHT peer to welcome trainers and record training statistics (e.g. loss, performance):
-   - In this example, we use [wandb.ai](https://wandb.ai/site) to plot training metrics; If you're unfamiliar with Weights & Biases, here's a [quickstart tutorial](https://docs.wandb.ai/quickstart).
-   - Run `python run_training_monitor.py --experiment_prefix NAME_YOUR_EXPERIMENT --wandb_project WANDB_PROJECT_HERE`
-   - `NAME_YOUR_EXPERIMENT` must be a unique name of this training run, e.g. `my-first-albert`. It cannot contain `.` due to naming conventions.
-   - `WANDB_PROJECT_HERE` is a name of wandb project used to track training metrics. Multiple experiments can have the same project name.
+
+### First peer
```
Comment (yhn112) [inline examples/albert/README.md:17]: I hate `whatsmyip` as a dependency.
Let's just say "if you don't know your public IP address, run `curl ifconfig.me`" or something like that

**Review by yhn112 — APPROVED** [review]

File: examples/albert/README.md (hunk 7 above)
Comment (mryab) [reply examples/albert/README.md:17]: Replaced with `requests.get`

**Pull request merged**